File: trading-platform/test_orderbook.go
package main

import (
	"fmt"
	"net/http"
)

func main() {
	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		w.WriteHeader(http.StatusOK)
		w.Write([]byte(`{"status":"ok"}`))
	})
	
	fmt.Println("Orderbook Engine Initialized...")
	fmt.Println("Listening on 0.0.0.0:8080 - Waiting for orders...")
	http.ListenAndServe(":8080", nil)
}

File: trading-platform/test_orderbook.rs
use std::io::{Read, Write};
use std::net::TcpListener;

fn main() {
    println!("Orderbook Engine Initialized...");
    println!("Listening on 0.0.0.0:8080 - Waiting for orders...");

    let listener = TcpListener::bind("0.0.0.0:8080").expect("Failed to bind to 8080");

    for stream in listener.incoming() {
        match stream {
            Ok(mut stream) => {
                let mut buffer = [0; 1024];
                // Read the request so the client doesn't get connection reset
                let _ = stream.read(&mut buffer);
                
                let response = "HTTP/1.1 200 OK\r\nContent-Type: application/json\r\nConnection: close\r\nContent-Length: 15\r\n\r\n{\"status\":\"ok\"}";
                let _ = stream.write_all(response.as_bytes());
            }
            Err(_) => {}
        }
    }
}
